File: functions/liboqs/variants/c/0_fpr_half_3.c
#include <stdint.h>

static uint64_t fpr_half(uint64_t x) {
    if (x != 0) {
        x -= 1ULL << 52;
        if (((x >> 52) & 0x7FF) == 0) {
            return 0;
        }
    }
    return x;
}


int main(){
    fpr_half(0);
}

File: functions/liboqs/variants/go/0_fpr_half_3.go
package main

func fpr_half(x uint64) uint64 {
    if (x & (0x7FF << 52)) == 0 {
        return x >> 1
    }
    return (x & ((1 << 52) - 1)) | ((x - (1 << 52)) & (0x7FF << 52))
}


func main(){
    fpr_half(0)
}